aten/src/ATen/native/nested/NestedTensorUtils.h

```diff
@@ -400,7 +400,15 @@ inline Tensor wrap_tensor_node(
     }
     options = flat_tensors[0].options().merge_in(options_);
     nt_buffer = at::cat(flat_tensors);
+
+    // Temporarily disable meta if needed to ensure sizes are built as real tensors.
+    auto orig_keyset = c10::impl::tls_local_dispatch_key_set();
+    auto local_keyset = orig_keyset;
+    local_keyset.included_ =
+      local_keyset.included_.remove_backend(c10::BackendComponent::MetaBit);
+    c10::impl::_force_tls_local_dispatch_key_set(local_keyset);
     nt_sizes = at::native::stack(sizes);
+    c10::impl::_force_tls_local_dispatch_key_set(orig_keyset);
```

Comment:
Curiosity question: if `nt_sizes` is a tensor, is the idea that it can potentially live on cpu/cuda? Or is it always supposed to be cpu-side metadata

Reply:
it's currently always on cpu but we have plans to potentially cache the info on cuda https://github.com/pytorch/pytorch/issues/95815

Reply:
It is always a CPU-side tensor. There are alot of raw pointer access to this tensor that would break. Surpsingly that is not checked here: https://github.com/pytorch/pytorch/blob/32ffd70644d8593e0e126d4df62df27a61e23619/aten/src/ATen/NestedTensorImpl.cpp#L16

which is probably bug prone and should be fixed. 

That being said for some cuda operations that require NT size to be on device this transfer leads to syncs. I think as a group we are trying to map out what kernel implementations need what metadata - in what form and on which device. And with this knowledge we might consider adding/ changing the representation of the metadata that the NT tracks.